compiler/rustc_parse/src/parser/ty.rs

```diff
@@ -756,6 +756,46 @@ impl<'a> Parser<'a> {
             }
         }
 
+        // We could've continued to parse bounds but we found a type instead of a trait.
+        if !self.can_begin_bound() && self.check_type() {
+            let ty = self.parse_ty().unwrap();
+
+            let mut err = self.struct_span_err(ty.span, "expected a trait, found type");
+
+            let peeled = ty.peel_refs();
+            if peeled.span != ty.span {
+                // Just get the indirection part of the type.
+                let span = ty.span.with_hi(peeled.span.lo());
+
+                err.span_suggestion(
+                    span,
+                    "consider removing the indirection",
+                    "",
+                    Applicability::MaybeIncorrect,
+                );
+
+                match &peeled.kind {
+                    TyKind::Path(_, path) => {
```

Comment:
Minor nitpick regarding style: You could use `if`/`let` here which looks a bit nicer here in my opinion.

Reply:
Done. I was thinking that perhaps other cases could be added but on second thought none would work.